lib/Sema/CSDiagnostics.cpp

```diff
@@ -5943,9 +5936,8 @@ bool NonEphemeralConversionFailure::diagnosePointerInit() const {
                     ? diag::cannot_construct_dangling_pointer_warning
                     : diag::cannot_construct_dangling_pointer;
 
-  auto *anchor = getRawAnchor();
-  emitDiagnostic(anchor->getLoc(), diagID, constructedTy, constructorKind)
-      .highlight(anchor->getSourceRange());
+  emitDiagnostic(diagID, constructedTy, constructorKind)
+      .highlight(getSourceRange());
```

Comment:
Unfortunately we really do want to use the raw anchor here, as we want to highlight the entire constructor call for this specific diagnostic, as it's mainly talking about the invalid construction of the `UnsafePointer`.

Reply:
Fixed!